cwms-data-api/src/main/java/cwms/cda/api/location/kind/GateChangeGetAllController.java

```diff
@@ -0,0 +1,117 @@
+                            + "defaults to EN."),
+                    @OpenApiParam(name = PAGE_SIZE, type = Integer.class,
+                            description = "the maximum number of gate changes to retrieve, regardless of time window. " +
+                                    "A positive integer is interpreted as the maximum number of changes from the " +
+                                    "beginning of the time window. A negative integer is interpreted as the maximum number " +
+                                    "from the end of the time window. " + "Default " + DEFAULT_PAGE_SIZE + "." +
+                                    "A page cursor will not be returned by this DTO. Instead, the next page can be determined " +
+                                    "by querying the next set of changes using the last returned change date and using " +
+                                    "start-time-inclusive=false")
+            },
+            responses = {
+                    @OpenApiResponse(status = STATUS_200, content = {
+                            @OpenApiContent(from = GateChange.class, isArray = true, type = Formats.JSONV1)
+                    })
+            },
+            description = "Returns matching CWMS gate change data for a Reservoir Project.",
+            tags = {OutletController.TAG}
+    )
+    @Override
+    public void handle(@NotNull Context context) {
+        String office = context.pathParam(OFFICE);
+        String id = context.pathParam(PROJECT_ID);
+        CwmsId projectId = new CwmsId.Builder().withName(id).withOfficeId(office).build();
+        Timestamp startTime = Timestamp.from(requiredZdt(context, BEGIN).toInstant());
+        Timestamp endTime = Timestamp.from(requiredZdt(context, END).toInstant());
+        boolean startInclusive = context.queryParamAsClass(START_TIME_INCLUSIVE, Boolean.class).getOrDefault(true);
+        boolean endInclusive = context.queryParamAsClass(END_TIME_INCLUSIVE, Boolean.class).getOrDefault(false);
+        UnitSystem unitSystem = context.queryParamAsClass(UNIT_SYSTEM, UnitSystem.class).getOrDefault(UnitSystem.EN);
+        int pageSize = context.queryParamAsClass(PAGE_SIZE, Integer.class).getOrDefault(DEFAULT_PAGE_SIZE);
+
+        try (Timer.Context ignored = markAndTime(GET_ALL)) {
+            DSLContext dsl = JooqDao.getDslContext(context);
+            OutletDao dao = new OutletDao(dsl);
+            List<GateChange> changes = dao.retrieveOperationalChanges(projectId, startTime.toInstant(),
+                                                                      endTime.toInstant(), startInclusive, endInclusive,
+                                                                      unitSystem, pageSize);
+            String formatHeader = context.header(Header.ACCEPT) != null ? context.header(Header.ACCEPT) : Formats.JSONV1;
+            ContentType contentType = Formats.parseHeader(formatHeader, GateChange.class);
```

Comment:
My previous comment makes more sense here, than in POST, POST should probably error if it's not told at all.

Reply:
Mind if we resolve this comment as part of the one above?  Seems like that one has taken over this conversation.